### DISCUSSION TITLE
building on ubuntu 22.04.3 doesnt have syntax highlighting for Rust

### DISCUSSION THREAD

**Discussion by Corallus-Caninus** [helix-editor/helix#8951] [Troubleshooting]
### Summary

I followed the instructions exactly including setting up runtime files and fetching and building grammars but hx --health shows that there isnt any syntax highlighting. the only difference is I dont have lldb-vscode. I tried installing llvm as per some forum suggestions and linking the bin directory but it doesnt install lldb-vscode by default.

### Reproduction Steps

follow the build instructions exactly on ubuntu.

### Helix log

its empty.

### Platform

Ubuntu

### Terminal Emulator

GNOME

### Installation Method

cargo build

### Helix Version

latest (master)

**Comment by the-mikedavis** [comment]
lldb-vscode isn't necessary for syntax highlighting, just the runtime files. If you see all "X"s in `hx --health` then you haven't installed the runtime files properly.

**Reply by Corallus-Caninus** [reply]
i dont have all xs just lldb, highlighting etc. can anyone verify this or does ubuntu normally build highlighting correctly with fetch and build grammars?

**Comment by Corallus-Caninus** [comment]
I solved this by setting the runtime env to the snap binary after installing helix with apt:

bada@bada-System-Product-Name:~/Code/helix$ export HELIX_RUNTIME=/snap/helix/21/bin/runtime

it seems the instructions in the installation documentation https://docs.helix-editor.com/install.html#building-from-source for fetch and build don't setup highlighting on my system. 

I'll leave it up to the repo owners to close this since this doesnt seem like a proper solution

lldb-vscode is still not installed via "apt install llvm"

**Comment by jerabaul29** [comment]
I was bumping my head on the same issue (also on Ubuntu 22.04). The problem is how I was exporting the runtime location from my bashrc; I had added the line suggested in the documentation from https://docs.helix-editor.com/install.html#building-from-source :

```
HELIX_RUNTIME=~/src/helix/runtime
```

However, this was not working for me, I needed to add the following line instead:

```
export HELIX_RUNTIME="/home/kzm/src/helix/runtime"
```

I.e. I needed i) to make sure to ```export``` to pass it to child processes, and ii) to use the full name, i.e. ```/home/kzm``` instead of ```~```, using ```~``` was (weirdly?) resulting in a folder not found error. With this I get:

```
kzm@kzm-bpq:~/src/helix$ hx --health
Config file: /home/kzm/.config/helix/config.toml
Language file: default
Log file: /home/kzm/.cache/helix/helix.log
Runtime directories: /home/kzm/.config/helix/runtime;/home/kzm/src/helix/runtime;/home/kzm/.cargo/bin/runtime
Runtime directory is empty: /home/kzm/.cargo/bin/runtime
Clipboard provider: wl-paste+wl-copy
System clipboard provider: wl-paste+wl-copy

Language                LSP                     DAP                     Highlight               Textobject              Indent                  
agda                    None                    None                    ✓                       ✘                       ✘                       
bash                    ✘ bash-language-serv…   None                    ✓                       ✓                       ✓                       
[etc..., but looks as expected]
```

@Corallus-Caninus would be curious to hear if this fixes things for you too :) .

@the-mikedavis do you think it would make sense to update the documentation at https://docs.helix-editor.com/install.html#building-from-source from:

```
HELIX_RUNTIME=~/src/helix/runtime
```

to:

```
export HELIX_RUNTIME="/home/${USERNAME}/src/helix/runtime"
```

or something like this? :) (not sure why the ```~``` is not recognized, but adding the export does make sense for sure :) )

**Reply by jerabaul29** [reply]
(another thing I got when trying to build was the same issue as https://github.com/rust-lang/cc-rs/issues/909 , i.e. lots of issues along the line of:

```
cargo-warning:Compiler version doesn't include clang or GCC: "cc" "--version"
```

but somehow this is gone now, and it seems that these builds are cached somewhere, as if I try to re-run them, I just get

```
kzm@kzm-bpq:~$ hx --grammar fetch
Fetching 171 grammars
171 up to date git grammars
kzm@kzm-bpq:~$ hx --grammar build
Building 171 grammars
171 grammars already built
```

even trying to build from a new folder. So this seems not to have been the root cause of the issue (and the fix above does work for me).
